price-feeder/oracle/provider/kraken.go

```diff
@@ -23,100 +26,385 @@ type (
 	// KrakenProvider defines an Oracle provider implemented by the Kraken public
 	// API.
 	//
-	// REF: https://docs.kraken.com/rest/
+	// REF: https://docs.kraken.com/websockets/#overview
 	KrakenProvider struct {
-		baseURL string
-		client  *http.Client
+		wsURL             url.URL
+		wsClient          *websocket.Conn
+		logger            zerolog.Logger
+		muTickers         sync.Mutex
+		tickers           map[string]TickerPrice // Symbol => TickerPrice
+		muSubscribedPairs sync.Mutex
```

Comment:
typically we just call this `mtx` and have all the fields it protects below it :)

```suggestion
		mtx sync.Mutex
```

Reply:
So, we should use only one mutex to control both fields?

Reply:
yes, one `mtx` to control all concurrency protected fields :)

Reply:
done at https://github.com/umee-network/umee/pull/580/commits/d50b86651e10c827a31191e4754d57f339bef8d7